File: 16.py
import re
def text_analysis(an_string):
    if type(an_string)==str:
        an_string = an_string.lower()
        words = re.findall(r"(?<=\s)[a-záéíóúñ]+|^[a-záéíóúñ]+",an_string) #creamos una lista con las palabras, las cuales solo tienen letras de la a-z con acento o ñ
        #además, si empieza la cadena con la palabra, también es válido
        phrases = an_string.split(".") #separamos usando el punto para obtener las frases
        l=0 #longitud media
        m = "" #palabra más larga
        for word in words: #un solo bloque iterativo
            l+= len(word)/(len(words)) #sumamos a l la longitud de cada palabra dividida entre el total de palabras
            if len(m)<len(word):
                m=word #actualizamos la palabra más larga
        print(f"Palabras: {len(words)}     Frases: {len(phrases)}")
        print(f"longitud promedio de palabra: {l}   Palabra más larga: {m}")
text_analysis("Me siento muy contento, me siento muy feliz. Pues yo creo, me voy a divertir. Oahhh, oahhhhhhhh")

File: 16.js
"use strict"
const textAnalysis = text=>{
    if(typeof(text)=="string"){
        text = text.toLowerCase();
        let words = text.match(/(?<=\W|^)[a-záéíóúñ]+/g);
        let phrases = text.split(/\.+/);
        let m = "",l=0;
        for(let word of words){
            l+= (word.length)/(words.length);
            if(m.length<word.length){
                m=word;
            }
        }
        console.log(`Se obtuvo que la palabra más larga es ${m}, la longitud de palabra media es ${l} y hay ${phrases.length} frases`);
        return [m,l,phrases.length];
    }else{
        console.log("error, debes introducir una cadena");
        return null;
    }
}